src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,153 @@
 import { humanFileSize } from "util/helpers";
 import { limitToBytes } from "util/limits";
 import type { LxdProject } from "types/project";
 import { useServerEntitlements } from "util/entitlements/server";
+import { useQueries } from "@tanstack/react-query";
+import type { LxdResources } from "types/resources";
+import { Icon } from "@canonical/react-components";
+import type { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import type { CreateInstanceFormValues } from "pages/instances/CreateInstance";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { useClusterMembers } from "context/useClusterMembers";
+import { CLUSTER_GROUP_PREFIX } from "util/instances";
+import ResourceLink from "components/ResourceLink";
+import { useIsClustered } from "context/useIsClustered";
 
 interface Props {
+  formik?: InstanceAndProfileFormikProps;
   project?: LxdProject;
 }
 
-const MemoryLimitAvailable: FC<Props> = ({ project }) => {
+const MemoryLimitAvailable: FC<Props> = ({ formik, project }) => {
   const notify = useNotify();
   const { canViewResources } = useServerEntitlements();
+  const isClustered = useIsClustered();
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const values = formik?.values;
+  const projectResourceMemory = project?.config["limits.memory"];
+  const createTarget =
+    (values as CreateInstanceFormValues)?.target ?? undefined;
+  const currentServer = (() => {
+    if (createTarget && !createTarget.startsWith(CLUSTER_GROUP_PREFIX)) {
+      return createTarget;
+    }
+    const location = (values as EditInstanceFormValues)?.location ?? undefined;
+    return location || undefined;
+  })();
 
-  const {
-    data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
+  const clusterMemberNames = clusterMembers.map((member) => member.server_name);
+  const queries = useQueries({
+    queries: (isClustered ? clusterMemberNames : [currentServer])
+      .filter(
+        (member) => !isClustered || !currentServer || member === currentServer,
+      )
+      .map((member) => ({
+        queryKey: [queryKeys.resources],
+        queryFn: async () => fetchResources(isClustered ? member : ""),
+        enabled: canViewResources(),
+      })),
   });
+  const isLoading = queries.some((query) => query.isLoading);
+  const error = queries.find((query) => query.error)?.error;
+  const resources = queries
+    .map((query) => query.data)
+    .filter((data) => data !== undefined);
 
   if (isLoading) {
     return <Spinner className="u-loader" text="Loading resources..." />;
   }
-
   if (error) {
     notify.failure("Loading resources failed", error);
   }
-
-  const getAvailableMemory = () => {
-    if (!project?.config["limits.memory"]) {
-      return resources?.memory.total;
+  const getAvailableMemory = (resource: LxdResources) => {
```

Comment:
I think we should evolve the name a bit to make clear this function is for a targeted cluster member / unclustered case.